pkg/buildpack/builder.go

```diff
@@ -119,17 +123,52 @@ func (b *PackageBuilder) finalizeImage(image WorkableImage, tmpDir string) error
 	return nil
 }
 
-func (b *PackageBuilder) validate() error {
-	if b.buildpack == nil {
-		return errors.New("buildpack must be set")
+func (b *PackageBuilder) finalizeExtensionImage(image WorkableImage, tmpDir string) error {
+	if err := dist.SetLabel(image, MetadataLabel, &Metadata{
+		ModuleInfo: b.extension.Descriptor().Info(),
+	}); err != nil {
+		return err
 	}
 
-	if err := validateBuildpacks(b.buildpack, b.dependencies); err != nil {
+	exLayers := dist.ModuleLayers{}
+	exLayerTar, err := ToLayerTar(tmpDir, b.extension)
+	if err != nil {
 		return err
 	}
 
-	if len(b.resolvedStacks()) == 0 {
-		return errors.Errorf("no compatible stacks among provided buildpacks")
+	diffID, err := dist.LayerDiffID(exLayerTar)
+	if err != nil {
+		return errors.Wrapf(err,
+			"getting content hashes for extension %s",
+			style.Symbol(b.extension.Descriptor().Info().FullName()),
+		)
+	}
+
+	if err := image.AddLayerWithDiffID(exLayerTar, diffID.String()); err != nil {
+		return errors.Wrapf(err, "adding layer tar for extension %s", style.Symbol(b.extension.Descriptor().Info().FullName()))
+	}
+
+	dist.AddToLayersMD(exLayers, b.extension.Descriptor(), diffID.String())
+
+	if err := dist.SetLabel(image, dist.ExtensionLayersLabel, exLayers); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (b *PackageBuilder) validate() error {
+	if b.buildpack == nil && b.extension == nil {
+		return errors.New("buildpack or extension must be set")
+	}
+	if b.buildpack != nil && b.extension == nil {
+		if err := validateBuildpacks(b.buildpack, b.dependencies); err != nil {
```

Comment:
It might be worth throwing in a comment that we don't need to validate extensions because there is no order and there are no stacks

Reply:
Will do that